api/src/main/java/org/openmrs/hl7/handler/ORUR01Handler.java

```diff
@@ -1293,7 +1293,7 @@ private void updateHealthCenter(Patient patient, PV1 pv1) {
 			
 			PersonAttribute currentHealthCenter = patient.getAttribute("Health Center");
 			
-			if (currentHealthCenter == null || !currentHealthCenter.equals(newLocationId.toString())) {
+			if (currentHealthCenter == null || !newLocationId.toString().equals(currentHealthCenter.getValue())) {
 				PersonAttribute newHealthCenter = new PersonAttribute(healthCenterAttrType, newLocationId.toString());
 				
 				log.debug("Updating patient's location from " + currentHealthCenter + " to " + newLocationId);

```

Comment:
@sashrika by the way, do you remember how you got this? http://pastebin.com/0VaXVUDw